server/channels/jobs/base_workers.go

```diff
@@ -83,11 +83,14 @@ func (worker *SimpleWorker) DoJob(job *model.Job) {
 	c := request.EmptyContext(worker.logger)
 
 	var appErr *model.AppError
+	// We keep the reference to the old job because GetJob might return nil.
+	oldJob := job
 	// We get the job again because ClaimJob changes the job status.
 	job, appErr = worker.jobServer.GetJob(c, job.Id)
 	if appErr != nil {
```

Comment:
1/5 that it's simpler to write:
```golang
// We get the job again because ClaimJob changes the job status.
newJob, appErr := worker.jobServer.GetJob(c, job.Id)
if appErr != nil {
		job.Logger.Error("SimpleWorker: job execution error", mlog.Err(appErr))
		worker.setJobError(job, appErr)
}
job = newJob
```

This way, the error handling pattern stays the same.

Reply:
Ah yes, much better.